### PR TITLE
[Main] Adding new test cases for API Publisher

### PR DESCRIPTION
**Pull request by vishmi49**
Following test cases are added to the test suite

04-depricate-old-versions.spec.js
02-add-assign-global-scopes-for-api.spec.js
03-add-tags-for-the-api.spec.js
01-add-nested-comments-to-api.spec.js

### CODE REVIEW

File: tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import PublisherComonPage from "../../../support/pages/publisher/PublisherComonPage";
+const publisherComonPage = new PublisherComonPage();
+
+describe("Depricate old versions of api before publishing", () => {
+  const { publisher, password } = Utils.getUserInfo();
+  const apiName = Utils.generateName().replace("-", "_");
+  const apiVersion = "1.0.0";
+  const newVersion = "2.0.0";
+
+  const deployeandPublish = (apiId) => {
```
Comment (dulithsenanayake) [inline tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js:29]: Typo

File: tests/cypress/integration/publisher/002-api-resources/02-add-assign-global-scopes-for-api.spec.js [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+      cy.get("#resources-save-operations", { timeout: 30000 });
+      cy.get(`#${verb}\\${target}-operation-scope-select`)
+        .contains(scopeName)
+        .should("be.visible");
+
+        Utils.deleteAPI(apiId);
+    });
+
+
```
Comment (dulithsenanayake) [inline tests/cypress/integration/publisher/002-api-resources/02-add-assign-global-scopes-for-api.spec.js:104]: Remove unnecessary empty lines. Check others

File: tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+            .contains("DEPRECATED")
+            .should("exist");
+
+            Utils.deleteAPI(apiId);
+            Utils.deleteAPI(newAPIid);
+        });
+      }
+    );
+  });
+});
```
Comment (dulithsenanayake) [inline tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js:117]: Shall we add EOF here? Check others

**Review by dulithsenanayake — CHANGES_REQUESTED** [review]

File: tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js (hunk 3 above)
Comment (vishmi49) [reply tests/cypress/integration/publisher/011-lifecycle/04-depricate-old-versions.spec.js:120]: Resolved in the latest commit

**Review by dulithsenanayake — APPROVED** [review]

**Pull request merged**